fontc/src/workload.rs

```diff
@@ -228,8 +241,38 @@ impl<'a> Workload<'a> {
                         job.write_access.clone(),
                     );
 
+                    let abort = abort_queued_jobs.clone();
+
                     scope.spawn(move |_| {
-                        let result = work.exec(work_context);
+                        if abort.load(std::sync::atomic::Ordering::Acquire) {
+                            log::trace!("Aborting {:?}", work.id());
+                            return;
+                        }
+                        // # Unwind Safety
+                        //
+                        // 'unwind safety' does not impact memory safety, but
+                        // it may impact program correctness; the thread may have
+                        // left shared memory in an inconsistent state.
+                        //
+                        // I believe this is not a concern for us, as we cancel any
+                        // pending jobs after seeing a panic and jobs that depend
+                        // on state produced by the panicking job must be scheduled
+                        // after it. Unless we have jobs that are mutating
+                        // shared resources then I think this is fine.
+                        //
+                        // references:
+                        // <https://doc.rust-lang.org/nomicon/exception-safety.html#exception-safety>
+                        // <https://doc.rust-lang.org/std/panic/trait.UnwindSafe.html>
+                        let result = match std::panic::catch_unwind(AssertUnwindSafe(|| {
+                            work.exec(work_context)
+                        })) {
+                            Ok(result) => result,
+                            Err(err) => {
+                                let msg = get_panic_message(err);
+                                abort.store(true, std::sync::atomic::Ordering::Release);
```

Comment:
Strongly recommend importing Release (`abort.store(true, Release)`) rather than writing the fully qualified name; it's uninformative noise.

Reply:
Hm. Also are you quite sure Relaxed is insufficient? If so, why, and we should comment as much.